pkg/appender/ingest.go

```diff
@@ -78,12 +84,14 @@ func (mc *MetricsCache) metricFeed(index int) {
 					if app.metric == nil {
 						// Handle update completion requests (metric == nil)
 						completeChan = app.resp
-
 						length := mc.metricQueue.Length()
-						if gotCompletion && length == 0 && len(mc.asyncAppendChan) == 0 {
-							completeChan <- 0
-							gotCompletion = false
-							gotData = false
+						if length == 0 && len(mc.asyncAppendChan) == 0 {
```

Comment:
aren't you suppose to set `potentialCompletion` to `false` in the else of this condition?

Reply:
yep it needs to be set to false but it should be false in any case after we read next item from asyncChannel
if it's waitForComplition req then we are done and if not then set to false and move on